lib/node.js

```diff
@@ -175,13 +175,12 @@ Node.prototype.delete = function (key, distinct) {
 };
 
 Node.prototype._deleteRecursive = function (key, distinct, index) {
-  if (index === key.length || this.leaf) {
+  if (this.leaf) {
     // If the index matches the key length, we are at the node to be deleted.
     // Remove values associated with the key, where the `distinct` value also matches.
-    this.values = this.values.filter((item) => (item.key !== key && item.distinct !== distinct));
-
-    // If the node becomes a leaf node, update its status.
-    this.leaf = this.values.length > 0;
```

Comment:
Does the removal of this line cover this?
> stops errantly converting non-leaf nodes into broken "leaf" nodes when deleting.

Reply:
Yes exactly!